src/robotFactory.js

```diff
@@ -1,10 +1,66 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords = 0, chipVersion) {
```

Comment:
Keep the default data type same to normal.
```suggestion
  constructor(name, weight, coords = {}, chipVersion) {
```